Autosquash not triggered when the PR is validated

I have a personal repository where I mostly use PRs to run tests for me.
Therefore, every time I create a PR, I apply it the _autosquash_ label, to have it merge as soon as tests pass.
Currently, I see that this workflow does not get the PR merged as expected [1]. At the end of all checks, I can remove and set again the label. This time, as previous validations were ok, the action is triggered and the merge occurs.

I know it is not the initial goal of this action, but I can't see why it is not triggered, as it seems to be the purpose of the type "check_run:completed_.

[1] (https://github.com/Kineolyan/snow-hall/pull/7)
[2] (https://github.com/Kineolyan/snow-hall/pull/6/checks?check_run_id=370010823)

thanks @tibdex ! But when I run it on a schedule, I get this totally different error: 
```##[error]Cannot destructure property `name` of 'undefined' or 'null'.```

So I'd guess you extract some information from the github event that isn't there on a schedule obviously. The github_token I have already set, but I think I doesn't help the situation, that another gh action marks green.

--- a/.github/workflows/autosquash.yml
+++ b/.github/workflows/autosquash.yml
@@ -31,6 +31,9 @@ jobs:
         with:
           # We can't use the built-in secrets.GITHUB_TOKEN yet because of this limitation:
           # https://github.community/t5/GitHub-Actions/Triggering-a-new-workflow-from-another-workflow/td-p/31676
-          # In the meantime, use a personal access token with repo access.
-          # See https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line
+          # In the meantime, use a token granting write access on the repo:
+          # - a GitHub App token
+          #   See https://github.com/marketplace/actions/github-app-token.
+          # - a personal access token
+          #   See https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line.
           github_token: ${{ secrets.AUTOSQUASH_TOKEN }}

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "autosquash",
-  "version": "2.0.2",
+  "version": "2.1.0",
   "license": "MIT",
   "files": [
     "action.yml",

